### DISCUSSION TITLE
engine vs engine tournaments using winboard, problem with engines exiting unexpectedly

### DISCUSSION THREAD

**Discussion by musketeerchess** [fairy-stockfish/Fairy-Stockfish#779] [Analysis]
Hi,
I'm undergoing thorough tests using Fairy Stockfish. In most of the cases i play tournaments with the same engine (FS) playing against itself, TC 100 moves per minute, 2 CPU and 256 Mo Hash. I noticed many winboard crashes related to the engine exiting unexpectedly most of the time when the game finishes (whatever the result it doesn't matter) and very rarely during the course of games. 

What this could possibly be related to?
Thanks

**Comment by ianfab** [comment]
Hard to tell. It could be problems with the CECP implementation in Fairy-SF (assuming CECP is used and not UCI), or maybe simply a bug in WinBoard. Generally in my experience XBoard/WinBoard does not handle issues very gracefully, so there could be many possible reasons for a crash.

**Comment by musketeerchess** [comment]
Hi Fabian
Is there a methodology that can help sort out things?
The debug file of winboard just tells the engine exited. Using command
-xreuse lowers a ltttle bit the % of the crashes but doesn't resolve the
problem. Sometimes also the engine freezes with the time control becoming a
countdown and negative without the engine stopping or winboard stopping the
game by time forfeit.

Thanks for your help

Le mar. 23 avr. 2024 à 10:18, Fabian Fichter ***@***.***> a
écrit :
